src/apps/INTSC_HEXHEX.cpp

```diff
@@ -0,0 +1,277 @@
+  setVariantDefined( Lambda_CUDA );
+  setVariantDefined( RAJA_CUDA );
+
+  setVariantDefined( Base_HIP );
+  setVariantDefined( Lambda_HIP );
+  setVariantDefined( RAJA_HIP );
+}
+
+INTSC_HEXHEX::~INTSC_HEXHEX()
+{
+}
+
+
+void INTSC_HEXHEX::setUp(VariantID vid,
+                         Size_type RAJAPERF_UNUSED_ARG(tune_idx))
+{
+  // One standard intersection is 8 subzone intersections.
+  Index_type n_std_intsc  = getActualProblemSize() ;
+  Index_type n_subz_intsc = npairs_per_std_intsc * n_std_intsc ;
+
+  // coordinates for donor zone (the eight corner points)
+  Real_type xdzone[8] =
+      { m_xmin, m_xmax, m_xmin, m_xmax, m_xmin, m_xmax, m_xmin, m_xmax } ;
+
+  Real_type ydzone[8] =
+      { m_ymin, m_ymin, m_ymax, m_ymax, m_ymin, m_ymin, m_ymax, m_ymax } ;
+
+  Real_type zdzone[8] =
+      { m_zmin, m_zmin, m_zmin, m_zmin, m_zmax, m_zmax, m_zmax, m_zmax } ;
+
+  Real_type xtzone[8], ytzone[8], ztzone[8] ;
+  for ( Index_type i=0 ; i<8 ; ++i ) {
+    xtzone[i] = xdzone[i] + m_shift ;
+    ytzone[i] = ydzone[i] + m_shift ;
+    ztzone[i] = zdzone[i] + m_shift ;
+  }
+
+  auto a_ds = allocDataForInit ( m_dsubz, 24L*n_subz_intsc, vid ) ;
+  auto a_ts = allocDataForInit ( m_tsubz, 24L*n_subz_intsc, vid ) ;
+
+  //  Repeat the same calculation n_subz_intsc times, expand the
+  //  same donor and target zones.
+  for ( Index_type k=0 ; k < n_subz_intsc ; ++k ) {
+    for ( Index_type i=0 ; i<8 ; ++i ) {
+      m_dsubz[24L*k+ 0+i] = xdzone[i] ;
+      m_dsubz[24L*k+ 8+i] = ydzone[i] ;
+      m_dsubz[24L*k+16+i] = zdzone[i] ;
+      m_tsubz[24L*k+ 0+i] = xtzone[i] ;
+      m_tsubz[24L*k+ 8+i] = ytzone[i] ;
+      m_tsubz[24L*k+16+i] = ztzone[i] ;
+    }
+  }
+
+  const Int_type block_size = default_gpu_block_size ;
+  m_nthreads = tri_per_pair * n_subz_intsc ;  // 72 threads per subzone pair
+  Index_type gsize    = RAJA_DIVIDE_CEILING_INT(m_nthreads, block_size) ;
+
+  // intermediate volumes, moments
+  allocData ( m_vv_int, n_vvint_per_block * gsize, vid ) ;
+
+  allocAndInitDataConst ( m_vv_out, nvals_per_pair * n_subz_intsc, 0.0, vid ) ;
+
+  // output volumes and moments on the host
+  allocData ( m_vv, nvals_per_pair * n_subz_intsc, Base_Seq ) ;
+}
+
+
+//   Number of subzone intersections = 8 * number of standard intersections.
+//
+void INTSC_HEXHEX::check_intsc_volume_moments
+    ( Index_type const n_subz_intsc,  // number of subzone intersections
+      Real_const_ptr vv,   // computed volumes, moments on the host
+      VariantID vid )   // Print variant name in case of error
+{
+
+  {
+    Char_const_ptr tst = "INTSC_HEXHEX:" ;
+
+    //   Determine the correct volume and moments.
+    Real_type v0, vx, vy, vz ;
+
+    Real_type xmin = m_xmin, ymin = m_ymin, zmin = m_zmin ;
+    Real_type xmax = m_xmax, ymax = m_ymax, zmax = m_zmax ;
+
+    if ( m_shift > 0.0 ) {
+      xmin += m_shift ;   ymin += m_shift ;   zmin += m_shift ;
+    } else {
+      xmax -= m_shift ;   ymax -= m_shift ;   zmax -= m_shift ;
+    }
```

Comment:
```suggestion
    }
    
```

Blank lines between code chunks helps readability.

Reply:
https://github.com/LLNL/RAJAPerf/pull/554/commits/701606dafd8367bdd10cd48e0fd41f8360d10c17